src/libecalc/dto/components.py

```diff
@@ -167,11 +173,26 @@ class FuelConsumer(BaseConsumer):
     energy_usage_model: dict[Period, FuelEnergyUsageModel]
 
     _validate_fuel_consumer_temporal_models = field_validator("energy_usage_model", "fuel")(validate_temporal_model)
-
     _check_model_energy_usage = field_validator("energy_usage_model")(
         lambda data: check_model_energy_usage_type(data, EnergyUsageType.FUEL)
     )
 
+    def evaluate_emissions(
+        self,
+        energy_context: ComponentEnergyContext,
+        energy_model: EnergyModel,
+        expression_evaluator: ExpressionEvaluator,
+    ) -> Optional[dict[str, EmissionResult]]:
+        fuel_model = FuelModel(self.fuel)
+        fuel_usage = energy_context.get_fuel_usage()
+
+        assert fuel_usage is not None
```

Comment:
I would like us to have a discussion around assertions. I have always considered them bad in production code, but perhaps it is ok? I want to shed some light on that :)

Reply:
Sure! Let's discuss